iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -591,33 +591,60 @@ LogicalResult ReduceWindowOpConversion::apply(
     mhlo::ReduceWindowOp op, ArrayRef<Value> inputBuffers,
     ArrayRef<Value> resultBuffers, ConversionPatternRewriter &rewriter) const {
   auto loc = op.getLoc();
+  auto resultType = op.getResult().getType().cast<ShapedType>();
+  int rank = resultType.getRank();
+  if (rank != 4) {
+    return rewriter.notifyMatchFailure(op, "expected NHWC pooling-based op");
+  }
 
   // Create a fake window dimension.
   SmallVector<int64_t, 4> shapes;
-  for (auto dim : op.window_dimensions().getValues<int64_t>()) {
-    shapes.push_back(dim);
-  }
-  Type type = rewriter.getIntegerType(32);
-  auto memrefType = MemRefType::get(shapes, type);
+  shapes.push_back(op.window_dimensions().getValue<int64_t>(1));
+  shapes.push_back(op.window_dimensions().getValue<int64_t>(2));
+  auto memrefType = MemRefType::get(shapes, rewriter.getF32Type());
   auto fakeWindowDims = rewriter.create<AllocOp>(loc, memrefType);
 
-  llvm::SmallVector<Attribute, 4> strides;
-  if (op.window_strides().hasValue()) {
-    strides.insert(strides.begin(),
-                   op.window_strides().getValue().getAttributeValues().begin(),
-                   op.window_strides().getValue().getAttributeValues().end());
+  if (op.window_strides() &&
+      (op.window_strides().getValue().getValue<int64_t>(0) != 1 ||
+       op.window_strides().getValue().getValue<int64_t>(3) != 1)) {
+    return rewriter.notifyMatchFailure(
+        op, "expected window_strides to be [1,x,y,1]");
+  }
+  if (op.window_dimensions() &&
+      (op.window_dimensions().getValue<int64_t>(0) != 1 ||
+       op.window_dimensions().getValue<int64_t>(3) != 1)) {
+    return rewriter.notifyMatchFailure(
+        op, "expected window_dimensions to be [1,x,y,1]");
+  }
+
+  if (!inputBuffers[0].getType().cast<ShapedType>().getElementType().isF32()) {
+    return rewriter.notifyMatchFailure(op, "expected element type to be f32");
   }
-  auto stridesArg = ArrayAttr::get(op.getContext(), strides);
 
   // TODO(hanchung): Use template lambda after migrating to C++20.
+  Attribute strides;
```

Comment:
Is the above TODO still relevant? Don't know how long it will take us to migrate to C++20. :)

Reply:
Ah, this is for later lambda. yeah, let's remove it.